### PR TITLE
Payment Config: open Trustly in Dialog

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

- Open Trustly iFrame in a dialog instead of inline

- Refactor to be able to center the iFrame-Dialog on the screen

## Justify why they are needed

- Match how we handle payment connection in the mobile app

- Improve error handling and user experience

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #3444** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3444" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3444?utm_source=stack-comment).

File: apps/store/src/features/memberArea/PaymentsSection/PaymentConfig.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  )
+}
+
+const DialogIframeContent = styled(Dialog.Content)({
+  height: '100%',
+  maxHeight: 800,
+  width: '100%',
+  maxWidth: 600,
+  padding: theme.space.xs,
+})
```
Comment (gustaveen) [inline apps/store/src/features/memberArea/PaymentsSection/PaymentConfig.tsx:69]: Looks like the iframe isn't centered :) 
Screenshot 2023-10-17 at 14 34 07



```suggestion
  marginInline: 'auto',
})
```

File: apps/store/src/features/memberArea/PaymentsSection/PaymentConfig.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import { Button, Dialog, theme } from 'ui'
+import { useMemberAreaMemberInfoQuery, useTrustlyInitMutation } from '@/services/apollo/generated'
+import { useAppErrorHandleContext } from '@/services/appErrors/AppErrorContext'
+import { getTrustlyInitMutationVariables } from '@/services/trustly/createTrustlyUrl'
+import { TrustlyIframe } from '@/services/trustly/TrustlyIframe'
+import { useCurrentLocale } from '@/utils/l10n/useCurrentLocale'
+
+type State = { type: 'IDLE' } | { type: 'READY'; trustlyUrl: string }
+
+export const PaymentConfig = ({ startButtonText }: { startButtonText: string }) => {
```
Comment (gustaveen) [inline apps/store/src/features/memberArea/PaymentsSection/PaymentConfig.tsx:13]: nit: just something Malin said about the naming of this. In IEX they talk about it and communicate it as payment connection
```suggestion
export const PaymentConnection = ({ startButtonText }: { startButtonText: string }) => {
```

**Review by gustaveen — APPROVED** [review]
Super nice! 🙌

File: apps/store/src/features/memberArea/PaymentsSection/PaymentConfig.tsx (hunk 2 above)
Comment (robinandeer) [reply apps/store/src/features/memberArea/PaymentsSection/PaymentConfig.tsx:13]: Very good point, I just renamed from `PaymentConfiguration` but that is even better

File: apps/store/src/features/memberArea/PaymentsSection/PaymentConfig.tsx (hunk 1 above)
Comment (robinandeer) [reply apps/store/src/features/memberArea/PaymentsSection/PaymentConfig.tsx:69]: so strange, must be something up with dev env so it didn't sync my styles correctly

**Pull request merged**